src/FactSystem/FactControls/FactStepper.qml

```diff
@@ -0,0 +1,115 @@
+import QtQuick 2.5
+import QtQuick.Controls 1.2
+import QtQuick.Controls.Styles 1.2
+import QtQuick.Dialogs         1.2
+
+import QGroundControl.ScreenTools   1.0
+import QGroundControl.FactSystem 1.0
+import QGroundControl.Palette 1.0
+import QGroundControl.Controls 1.0
+
+Item {
+    property bool enabled: true
+    property alias fact: factInput.fact
+
+    property string valueText:          fact ? fact.valueString : "N/A"
+    property double factValue:          Number(valueText)
+
+    property double minimumValue:       !fact || isNaN(fact.min) ? 0 : fact.min
+    property double maximumValue:       !fact || isNaN(fact.max) ? 1 : fact.max
+    property double stepSize:           !fact || isNaN(fact.increment) ? 0 : fact.increment
+    property double stepRatio:          0.1
+
+    implicitHeight: 50
+    implicitWidth: implicitHeight * 4
+
+    function incrementWithScale(multiplier) {
+        var step = stepSize;
+        if (step == 0) {
+            step = Math.abs(stepRatio * factValue);
+        }
+
+        setFactValue(factValue + (step * multiplier));
+    }
+
+    function setFactValue(newValue) {
+        if (newValue < fact.min) {
+            newValue = fact.min;
+        }
+        if (newValue > fact.max) {
+            newValue = fact.max;
+        }
+
+        factValueChangeDelay.stop()
+        valueText = newValue.toLocaleString(Qt.locale(), 'f', fact.decimalPlaces)
+
+        console.log(fact.name, ": factValue=", factValue)
+        factValueChangeDelay.start()
+    }
+
+    Timer {
+        id: factValueChangeDelay
+
+        interval: 250
+        repeat: false
+        running: false
+
+        onTriggered: {
+            fact.value = factValue;
+            fact.valueChanged(fact.value)
+        }
+    }
+
+    Row {
+        anchors.fill:   parent
+
+        QGCButton {
+            id: minusButton
+            width : parent.width / 4
+            height: parent.height
+            text: "-"
+            enabled: fact ? (factValue > fact.min) : false
+
+            onClicked: {
+                incrementWithScale(-1)
+            }
+        }
+
+        QGCTextField {
```

Comment:
Since you not only support using +- buttons but you also support typing in a value you need to validate that values that are entered in there. This is why FactTextField shows right side dialogs if you enter in values which fail the validation specified by the parameter meta data. This isn't a mobile or desktop difference, this is just basic field validation which is a required thing to keep parameters values clean.